streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1125,6 +1125,13 @@ private void completeShutdown(final boolean cleanRun) {
         } catch (final Throwable e) {
             log.error("Failed to close restore consumer due to the following error:", e);
         }
+        if (producer != null) {
```

Comment:
We actually close the thread producer within `TaskCreator#close()` that is called within `taskManager.shutdown()` that is called in L114 above.

Is there a problem I might overlook?